build/build.ps1

```diff
@@ -87,8 +82,52 @@ function LocateVisualStudio {
     [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
     Invoke-WebRequest "http://github.com/Microsoft/vswhere/releases/download/$vswhereVersion/vswhere.exe" -OutFile $vswhereExe
   }
+  
+  return $vsWhereExe
+}
+
+function GetVSIXExpInstallerExe{
+  $vsixExpInstallerVersion = GetVersion("VSIXExpInstallerVersion")
+  $vsixExpInstalleDir = Join-Path $ToolsRoot "VSIXExpInstaller\$vsixExpInstallerVersion"
+  $vsixExpInstalleExe = Join-Path $vsixExpInstalleDir "tools\VSIXExpInstaller.exe"
+  
+  if (!(Test-Path $vsixExpInstalleExe)) {
+    Create-Directory $vsixExpInstalleDir
+    Write-Host "Downloading VSIXExpInstaller"
+    [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+    Invoke-WebRequest "https://dotnet.myget.org/F/roslyn-tools/api/v2/package/RoslynTools.VSIXExpInstaller/$vsixExpInstallerVersion" -OutFile RoslynTools.VSIXExpInstaller.zip
+    Expand-Archive .\RoslynTools.VSIXExpInstaller.zip -DestinationPath $vsixExpInstalleDir
+  }
+  
+  return $vsixExpInstalleExe
+}
+
+function GetTRXUnitExe{
+  $trxUnitVersion = GetVersion("TRXUnitVersion")
+  $trxUnitDir = Join-Path $ToolsRoot "TRXUnit\$trxUnitVersion"
+  $trxUnitExe = Join-Path $trxUnitDir "tools\TRXunit.exe"
+  if (!(Test-Path $trxUnitExe)) {
+    Create-Directory $trxUnitDir
+    Write-Host "Downloading TRXUnit"
+    [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+    Invoke-WebRequest "https://dotnet.myget.org/F/roslyn-tools/api/v2/package/TRXunit/$trxUnitVersion" -OutFile TRXUnit.zip
+    Expand-Archive .\TRXUnit.zip -DestinationPath $trxUnitDir
```

Comment:
Same comment as above.

Reply:
I think Roslyn just has a dummy `.proj` with a set of `<PackageReference>` elements that it calls restore on directly to get the packages on the machine in the right folder.  That's what I was suggesting instead of manually handling it.  That would get benefits such as verifying shas and signatures, proxy handling, etc.  

@jaredpar - is there an example in Roslyn?

Reply:
Indeed we essentially have a Roslyntoolset project just to download all of the tools we need. It's the first thing we restore. 

https://github.com/dotnet/roslyn/blob/master/build/ToolsetPackages/RoslynToolset.csproj

Reply:
alright, after wasting several hours on this: we can't do this until we move to using dotnet sdk for builds (presumably as part of updating repotoolset).  Using the MSBuild restore task does not appear to place packages in a predictable location.

Reply:
Can you elaborate?

Reply:
What different locations are they choosing? Using `msbuild /t:restore` should work equivalently at this point to `dotnet restore` in terms of where packages go (as long as you're using `PackageReference`). I definitely hit this type of problem back in the `nuget restore` days but haven't since we moved to msbuild based restore.

Reply:
I am reinstalling VS on my machine to see if it was in some bad state. This is what I observed:

0. I updated the version of `VSIXExpInstaller`
1. Restored the toolset project with package refs to `VSIXExpInstaller`
2. nuget said project restored correctly and even with diag set I sound no output indicating an error had occurred
3. `%USERPROFILE%\.nuget\packages\roslyntools.vsixexpinstaller\` contained no new version only the old one.

Reply:
Yep it still repros after reinstallation:

Version.props
```XML
...
<RoslynToolsVsixExpInstallerVersion>1.0.0-beta-62503-02</RoslynToolsVsixExpInstallerVersion>
...
```
Toolset.proj
```XML
...
<PackageReference Include="RoslynTools.VsixExpInstaller" Version="$(RoslynToolsVsixExpInstallerVersion)" />
...
```


build.ps1
```ps1
...
$ToolsetRestoreProj = Join-Path $PSScriptRoot "Toolset.proj"
...
function InstallToolset {
  if (!(Test-Path $ToolsetBuildProj)) {
    & $MsbuildExe $ToolsetRestoreProj /t:restore /m /nologo /clp:None /warnaserror /v:quiet /p:NuGetPackageRoot=$NuGetPackageRoot /p:BaseIntermediateOutputPath=$ToolsetDir /p:ExcludeRestorePackageImports=true
  }
}
```

And nothing gets put in `%USERPROFILE%\.nuget\packages\roslyntools.vsixexpinstaller`

Reply:
**NOTE:** This should repro on master on anyone's machine. Restoring `VsixExpInstaller` should already be happening), but it isn't.

https://github.com/dotnet/project-system/blob/b558ffacd4d6871d15362a9c0122b377cde55c6c/build/Versions.props#L15

https://github.com/dotnet/project-system/blob/b558ffacd4d6871d15362a9c0122b377cde55c6c/build/Toolset.proj#L13

https://github.com/dotnet/project-system/blob/b558ffacd4d6871d15362a9c0122b377cde55c6c/build/build.ps1#L100-L104

Reply:
@tannergooding have yo useen this before?

Reply:
What does the assets file look like? Does it have one?